### DISCUSSION TITLE
hardware encoding must use uid=root?

### DISCUSSION THREAD

**Discussion by GavinIves** [photoprism/photoprism#3738] [Q&A] [answered]
In docker, when I enable Intel's core display for hardware encoding, my uid must be set to root, otherwise hardware decoding cannot be performed. Will this cause security issues? Or is there something wrong with my configuration?
<img width="366" alt="image" src="https://github.com/photoprism/photoprism/assets/38970877/6b3669e9-4791-4e9f-9689-654764f814c0">
<img width="691" alt="image" src="https://github.com/photoprism/photoprism/assets/38970877/74b6df0f-d49a-44cf-be14-84b28f75bdf7">
but i use root, its ok!!!
<img width="373" alt="image" src="https://github.com/photoprism/photoprism/assets/38970877/75cbd88e-54d1-41de-83e9-600baae82041">
<img width="794" alt="image" src="https://github.com/photoprism/photoprism/assets/38970877/0c155142-c99f-418b-a2c5-d59b9b971114">

maybe like emby,use GIDLIST # A comma-separated list of additional GIDs to run emby as (default: 2) to solve it
url: https://hub.docker.com/r/emby/embyserver
<img width="806" alt="image" src="https://github.com/photoprism/photoprism/assets/38970877/a2549f9e-1f61-4560-8f0d-05b4e7f84e13">
<img width="775" alt="image" src="https://github.com/photoprism/photoprism/assets/38970877/2e1808c5-9a86-49ed-b251-b21c14703f7f">

**Comment by lastzero** [comment]
As an alternative, you can try to run the PhotoPrism service as a non-root user by setting the PHOTOPRISM_UID [environment variable](https://docs.photoprism.app/getting-started/config-options/#docker-image) in your `docker-compose.yml` file:

https://docs.photoprism.app/getting-started/advanced/docker-security/#run-services-as-non-root-user

It may also help to adjust the permissions of the related device:

https://docs.photoprism.app/getting-started/advanced/transcoding/#intel-quick-sync

If we need to change something in our Docker image to make it compatible with your specific environment, please let us know what exactly is needed. Thank you!

**Reply by GavinIves** [reply]
i had ran the PhotoPrism service as a non-root user by setting the PHOTOPRISM_UID, but when i use non-root,hardware encoding will invalid, like my pictures shows

**Reply by lastzero** [reply]
See the `create-users.sh` script for the device groups and users currently supported:

https://github.com/photoprism/photoprism/blob/e7370838221c373a23dc9a09f186a6065e2a5162/scripts/dist/create-users.sh#L13-L30

We've decided to go with that approach as you otherwise always have to start the container as root and can ONLY use the PHOTOPRISM_UID variable to change the UID later on. However, this means we might need to perform changes - for example add a new group - to allow using hardware transcoding on all devices & operating systems. In addition, the group ID 100 doesn't seem to be [officially supported](https://docs.photoprism.app/getting-started/config-options/#docker-image) (supported GID ranges are 0, 33, 50-99, 500-600, and 900-1200 in addition to the GIDs shown in the script above).

What about you use GID 109? Will this solve your issue?

I also recommend checking related GitHub issues, for example:
- #2739
- #2228

**Reply by GavinIves** [reply]
<img width="780" alt="image" src="https://github.com/photoprism/photoprism/assets/38970877/862e9a29-09aa-4b1d-8513-34250cf710e1">
uid=1100 gid=100 is supported. The official government just does not support uid. It is outside the range you mentioned. You remembered it wrongly. Aren't the uid and gid written here supposed to correspond to the permissions of my photo directory? So I need uid=1100 gid=100 gid=109 is the permission required to call renderd, and 44 is the permission to call video. Looking at the sh you sent, we can tell that he should have been added to the permissions, but I know that if uid=0 is not set, hardware decoding is unavailable. This is a very strange problem. I hope you can test it.

**Reply by lastzero** [reply]
I've spent the entire morning updating our docs to provide more specific guidance on Intel Video Transocoding and the UID/GID settings:

https://docs.photoprism.app/getting-started/advanced/transcoding/#intel-quick-sync

You are correct that the group 100 ("users") exists by default in the base Ubuntu image we are using. However, our script does not create it, so it may not be available in future base images, although it is unlikely that the Ubuntu maintainers will change this any time soon.

**Since we have a long list of other requests and tasks to take care of so we can release a new stable version, this is all the help I can provide at this time. If you need another group to be added to our image, feel free to let us know so we can include it either in the next release or the release after that (next month).**

**Reply by GavinIves** [reply, marked as answer]
Thank you very much. I understand what you mean. When we need to use the render, we must first obtain the gid of the render. ```getent group render | cut -d: -f3```
Using the command on the host machine will return the gid. Just set our gid to it. I set the gid to 109 and successfully achieved the goal.Thanks again

**Reply by dmkiraly** [reply]
This command returned "107" on my host machine, which is not on the supported list. What do I do then?
